chapters/en/slides/module7/module7_02.Rmd

```diff
@@ -0,0 +1,342 @@
+by binding `selection_multi` to it.
+
+We then need to link this new selection
+(called `click_legend` to distinguish it from the previous slide)
+to the points opacity encoding and add it to the plot.
+
+Now if you click the legend,
+it will select the corresponding observations in the scatterplot!
+
+---
+
+## Legend selections can be added to multiple charts
+
+```{python}
+bars = bars.encode(opacity=alt.condition(click_legend, alt.value(0.9), alt.value(0.2)))
+(points & bars).add_selection(click_legend) 
+```
+
+Notes:
+When creating a combined charts the syntax 
+for adding the legend interactivity is similar,
+but we need to add the selection to both charts
+since the legend belongs to both of them.
+
+We could do this one by one,
+or both at once as we do in this slide.
+
+---
+
+## Instead of highlighting, a selection can be used for filtering data
+
+```{python}
+bars = bars.transform_filter(brush)
+(points & bars).add_selection(click_legend) 
+```
+
+Notes:
+So far we have seen how to change graphical encodings such as color and opacity
+to highlight points in the data.
+Another useful type of interactivity is to actually filter the data
+based on a selection,
+rather than just styling the chart appearance.
+
+For example,
+we might wonder how many cars from different regions
+that we have selected when we drag in the scatter plot.
+Let's change the bar chart to update dynamically 
+as we select points in the scatterplot.
+
+We can do this by adding a transform filter to our bar plot.
+Transform filters can filter data similarly to what we might do in pandas,
+but are not as powerful in general
+so they are best used for operations
+like interactive filtering which are not possible in pandas.
+This type of interaction is sometimes called a dynamic query.
+
+We're keeping the legend here since we have conditions 
+in the charts' encodings that depend on the `legend_click` selection variable,
+but we could remove it if we change these.
+
+Now that we have select points by dragging in the scatter plot
+we can see that the barchart updates the count
+to reflect the number of points we have selected of each color,
+neat!
+
+---
+
+## Fixing the extent of the x-axis makes the bars easier to compare
+
+```{python}
+bar_chart_max = cars.groupby('Origin').size().max()
+bars = bars.encode(alt.X('count()', scale=alt.Scale(domain=[0, bar_chart_max])))
+(points & bars).add_selection(click_legend) 
+```
+
+Notes:
+It is a bit non-intuitive that the x-axis keep changing its extent
+as we select different number of points,
+and it makes the bar heights harder to compare
+as we move the selection around.
+Ideally only the bars would change length,
+but the axis would stay the same.
+
+In this slide we calculate the max number of cars
+from any region
+and then use this number to fix the extent of the x-axis.
+Now that we select in the scatterplot,
+only the lengths of the bars changes,
+but the range of the axis remains the same.
+
```

Comment:
What about fixing the number of categories on the y-axis? Should we do that too? It might make it easier, especially in plots with more than 3 categories...

Reply:
I don't think this is trivial,otherwise would be good.